### PR TITLE
Adds route naming to improve usability.

### PR DESCRIPTION
**Pull request by jcchavezs**
This PR allows to use the route path as span name without injecting the `router` as it degrades significantly the performance.

For more info read https://github.com/jcchavezs/zipkin-symfony/pull/6/files#diff-04c6e90faac2675aa89e2176d2eec7d8R144

### CODE REVIEW

**Comment by jcchavezs** [comment]
FYI https://github.com/openzipkin/brave/pull/602

File: src/ZipkinBundle/SpanNaming/Route/Naming.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function getName(Request $request)
+    {
+        $routeName = $request->request->get('_route');
+
+        if (array_key_exists($routeName, $this->routes)) {
```
Comment (adriancole) [inline src/ZipkinBundle/SpanNaming/Route/Naming.php]: fyi in brave we default to "get not_found" or "get redirected" when we are in a framework that supports routes, but there was no route found.

File: src/ZipkinBundle/Middleware.php [hunk 2]
```diff
@@ -60,7 +69,7 @@ public function onKernelRequest(GetResponseEvent $event)
 
         $span = $this->tracing->getTracer()->nextSpan($spanContext);
         $span->start();
-        $span->setName($request->getMethod());
+        $span->setName($this->spanNaming->getName($request));
```
Comment (jcchavezs) [inline src/ZipkinBundle/Middleware.php]: This should run terminate.

File: src/ZipkinBundle/SpanNaming/Route/Naming.php (hunk 1 above)
Comment (jcchavezs) [reply src/ZipkinBundle/SpanNaming/Route/Naming.php]: This is not the same case. If `$routeName` is null then we return a `not_found`. If `$routeName != null` and it is not in the `routes` then there is something wrong and we fail gracefully.

**Comment by adriancole** [comment]
The intent of not_found or redirected (in span name) is that someone
crawling or DOS attack wont fill up span namespace. And you can easily
search for this by name. So maybe a test of what happens when you get 404
is helpful.

**Comment by jcchavezs** [comment]
@adriancole sure, https://github.com/jcchavezs/zipkin-symfony/pull/6/files#diff-759dcfcf030a5d11ef05f571c336e998R16 covers the case where the request does not match any route.

**Pull request merged**